File: VirtualWorld-Python/rosliny/Barszcz.py
from Roslina import Roslina

class Barszcz(Roslina):
    
    def __init__(self):
        pass

    def __init__(self, swiat, poz, wiek, inicj, moc):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeWiek(wiek)
        super().changeInicjatywa(inicj)

        s = "B"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    def __init__(self, swiat, poz):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeInicjatywa(0)
        super().zmienSile(10)
        super().changeWiek(0)

        s = "B"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    #--------------------------------------------

    def rysowanie(self):
        print('B')

    def getZnak(self):
        return 'B'

    def isRadioactive(self):
        return True

    def czyZabojcza(self):
        return True

File: VirtualWorld-Cpp/Virtual-World/Barszcz.cpp
#include "Owca.h"
#include "Zwierze.h"
#include <iostream>
#include "Organizm.h"
#include "Trawa.h"
#include "Roslina.h"
#include "Jagody.h"
#include "Barszcz.h"
#include "DEF.h"

Barszcz::Barszcz()
{
}

Barszcz::Barszcz(Swiat* swiat, std::vector<unsigned int> poz, int wiek, int inicj, int moc)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeWiek(wiek);
	changeInicjatywa(inicj);

	getSwiat()->edytujPlansze(poz[0], poz[1], BARSZCZ);
}

Barszcz::Barszcz(Swiat* swiat, std::vector<unsigned int> poz)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeInicjatywa(0);
	zmienSile(10);
	changeWiek(0);

	getSwiat()->edytujPlansze(poz[0], poz[1], BARSZCZ);
}

void Barszcz::rysowanie()
{
	std::cout << BARSZCZ;
}

char Barszcz::getZnak()
{
	return BARSZCZ;
}

bool Barszcz::czyZabojcza()
{
	return 1;
}

bool Barszcz::isRadioactive()
{
	return 1;
}

Barszcz::~Barszcz()
{
}